src/app/loadout-drawer/loadout-item-conversion.ts

```diff
@@ -0,0 +1,47 @@
+import { D1ManifestDefinitions } from 'app/destiny1/d1-definitions';
+import { D2ManifestDefinitions } from 'app/destiny2/d2-definitions';
+import { InventoryBuckets } from 'app/inventory/inventory-buckets';
+import { makeFakeItem } from 'app/inventory/store/d2-item-factory';
+import { emptyArray } from 'app/utils/empty';
+import { DimItem } from '../inventory/item-types';
+import { DimLoadoutItem, LoadoutItem } from './loadout-types';
+import { findItem } from './loadout-utils';
+
+/**
+ * Turn the loadout's items into real DIM items. Any that don't exist in inventory anymore
+ * are returned as warnitems.
+ */
+export function getItemsFromLoadoutItems(
+  loadoutItems: LoadoutItem[] | undefined,
+  defs: D1ManifestDefinitions | D2ManifestDefinitions,
+  buckets: InventoryBuckets,
+  allItems: DimItem[]
+): [DimLoadoutItem[], DimItem[]] {
+  if (!loadoutItems) {
+    return [emptyArray(), emptyArray()];
+  }
+
+  const items: DimLoadoutItem[] = [];
+  const warnitems: DimItem[] = [];
+  for (const loadoutItem of loadoutItems) {
+    const item = findItem(allItems, loadoutItem);
```

Comment:
I just had a look over `findItems` and we are iterating over all inventory items for each item in the loadout. I wonder if its worth having some kind of map from item id to `DimItem` instead. We maximum items is 500ish so it's probably not a big deal.

Reply:
Yeah it could be quite a bit more efficient if we wanted it to be.